src/easier68k/core/opcodes/move.py

```diff
@@ -0,0 +1,216 @@
+from ...core.enum.ea_mode import EAMode
+from ...core.enum.op_size import MoveSize
+from ...core.enum.ea_mode_bin import EAModeBinary
+from ...simulator.m68k import M68K
+from ...core.opcodes.opcode import Opcode
+
+
+class Move(Opcode):
+    pass
+
+
+class Move(Opcode):
+    # Allowed values: nothing, or some combination of B, W, and L (for byte, word, and long)
+    # For example, MOVE would have 'BWL' because it can operate on any size of data, while MOVEA would have 'WL' because
+    # it can't operate on byte-sized data
+    allowed_sizes = 'BWL'
+
+    def __init__(self, command: str, parameters: str):
+        """
+        >>> str(Move('MOVE.B', '-(A0), D1'))
+        'Move command: Size B, src Mode: 4, Data: 0, dest Mode: 0, Data: 1'
+
+        >>> str(Move('MOVE.L', 'D3, (A0)'))
+        'Move command: Size L, src Mode: 0, Data: 3, dest Mode: 2, Data: 0'
+
+        >>> Move('MOVE.W', 'D3, A3')
+        Traceback (most recent call last):
+        ...
+        AssertionError
+
+        :param command:
+        :param parameters:
+        """
+        parts = command.split('.')  # Split the command by period to get the size of the command
+        assert len(parts) <= 2  # If we have more than 2 parts something is seriously wrong
+        if len(parts) == 1:  # Use the default size
+            size = 'W'
+        else:
+            assert parts[1] in Move.allowed_sizes, "Size {} isn't allowed for command {}".format(parts[1], command[0])
+            size = parts[1]
+
+        # Split the parameters into EA modes
+        params = parameters.split(',')
+        assert len(params) == 2
+
+        src = EAMode.parse_ea(params[0].strip())  # Parse the source and make sure it parsed right
+        assert src.mode > EAMode.ERR, 'Error parsing src'  # -1 means error
+
+        dest = EAMode.parse_ea(params[1].strip())
+        assert dest.mode > EAMode.ERR, 'Error parsing dest'  # -1 means error
+
+        Move.__init__(self, src, dest, size)
+
+    # Include parameters here
+    def __init__(self, src: EAMode, dest: EAMode, size='Ws'):
+        # Check that the src is of the proper type (for example, can't move from an address register for a move command)
+        assert src.mode != EAMode.ARD  # Only invalid src is address register direct
+        self.src = src
+
+        # Check that the destination is of a proper type
+        assert dest.mode != EAMode.ARD and dest.mode != EAMode.IMM  # Can't take address register direct or immediates
+        self.dest = dest
+
+        # Check that this is a valid size (for example, 'MOVEA.B' is not a valid command)
+        assert size in Move.allowed_sizes
```

Comment:
need to ensure that this works with both upper and lower case

Reply:
Will do